pkg/rpc/server/server.go

```diff
@@ -79,12 +87,34 @@ func (s *StoreServer) GetBlock(
 	pbData := data.ToProto() // Assuming data.ToProto() exists and doesn't return an error
 
 	// Return the successful response
-	return connect.NewResponse(&pb.GetBlockResponse{
+	resp := &pb.GetBlockResponse{
 		Block: &pb.Block{
 			Header: pbHeader,
 			Data:   pbData,
 		},
-	}), nil
+	}
+
+	// Fetch and set DA heights
+	rollkitBlockHeight := header.Height()
+	if rollkitBlockHeight > 0 { // DA heights are not stored for genesis/height 0 in the current impl
+		headerDAHeightKey := fmt.Sprintf("%s/%d/h", store.RollkitHeightToDAHeightKey, rollkitBlockHeight)
```

Comment:
@alpe taught me about `ds.NewKey(blockResponseKey).ChildString(foo)` Is it relevant here?